portfolio/src/main/java/com/google/sps/servlets/CommentService.java

```diff
@@ -16,45 +16,63 @@
+import com.google.appengine.api.images.ServingUrlOptions;
+import java.util.Map;
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.Sentiment;
+import com.google.cloud.language.v1.ClassifyTextRequest;
+import com.google.cloud.language.v1.ClassifyTextResponse;
+import com.google.cloud.language.v1.ClassificationCategory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
 
-/* a class with useful functions for regular comments */
+/* a class with useful functions for comments */
 public class CommentService {
     private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
-    /* find and return comments in descending order by time and return the 
+    /* find and return blog comments in descending order by time and return the 
     first num numbers */
-    public List<Entity> findAllComments(int num, String sort) {
-        Query query;
-        // select sorts based on input
-        if (sort.equals("time-desc"))
-            query = new Query("Comment").addSort("time", SortDirection.DESCENDING);
-        else if (sort.equals("time-asc"))
-            query = new Query("Comment").addSort("time", SortDirection.ASCENDING);
-        else if (sort.equals("content-asc"))
-            query = new Query("Comment").addSort("content", SortDirection.ASCENDING);
-        else
-            query = new Query("Comment").addSort("name", SortDirection.ASCENDING);
-        PreparedQuery results = datastore.prepare(query);
-        if (num == -1)
-          return results.asList(FetchOptions.Builder.withDefaults());
-        else
-          return results.asList(FetchOptions.Builder.withLimit(num));
-    }
+    public List<Entity> findAllComments(int num, int id, String sort, boolean isBlog) {
+      // create filter
+      Filter keyFilter = new FilterPredicate("postid", FilterOperator.EQUAL, id);
 
-    /* find and return the comment with the specific id */
-    public Entity findSpecificComments(long id) {
-        // create filter
-        Filter keyFilter = new FilterPredicate("id", FilterOperator.EQUAL, id);
-        Query query = new Query("Comment").setFilter(keyFilter);
-
-        PreparedQuery results = datastore.prepare(query);
-        List<Entity> list = results.asList(FetchOptions.Builder.withDefaults());
-        if (list.isEmpty()) {
-            System.err.println("EMPTY" + id);
-            return null;
-        } else {
-            return list.get(0);
-        }
+      // create query
+      Query query;
+      if (isBlog) {
+          query = new Query("PostComment").setFilter(keyFilter);
+      } else {
+          query = new Query("Comment");
+      }
+      // select sorts based on input
+      if (sort.equals("time-desc"))
```

Comment:
The sorting also repeats between blog and comment; this can go into a shared utility for adding sort to a query.

Reply:
I'm not sure I understand what you mean. This sorting code is currently a shared piece of code.

Reply:
I think I was confused, sorry!